compiler/ballerina-lang/src/main/java/io/ballerina/projects/Project.java

```diff
@@ -87,6 +92,11 @@ protected void setCurrentPackage(Package currentPackage) {
     }
 
     public ProjectEnvironment projectEnvironmentContext() {
```

Comment:
How do we ensure that this method is invoked right after a compiler crash? 

There can be places where we have cached compilerContext instances. If that's the case, then we will end up having multiple stale instances of compilerContext objects.

Reply:
Refactored the implementation as follows:
* Removed the crash recovery part from the Project API (which was added in the initial commit)
* Improved `Project.clearCaches` to reset the entire environment

The API user can call the `Project.clearCaches` method when there is a crash.